packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/findRootSpanId.tsx

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Span } from '../../../../types/trace';
+
+const findRootSpanId = (spans: Span[]): string | undefined => {
```

Comment:
packages/jaeger-ui/src/model/transform-trace-data.tsx : `transformTraceData` already constructs a tree of spans. Why do we need this re-implemented?

Reply:
As we need to start critical path algorithm from root span id.
And practically we can directly assign `rootSpanId = spans[0].spanId` as all spans are sorted based on startTime in the trace data.
So what's your view can I directly assign rootSpanId or use findRootSpanId function.

Reply:
My concern is that you're redoing the work that was already done on a trace after it is loaded. For large traces this could introduce a lot of additional overhead, especially if you're using linear searches and the CP logic is always invoked.

Reply:
Yeah I got your point, I removed findRootSpanId function as it is already assured that first span will be always a rootSpan
https://github.com/jaegertracing/jaeger-ui/blob/ad932aa4fdc20a22f8619aeaef5d541f386bd45a/packages/jaeger-ui/src/components/TracePage/CriticalPath/index.tsx#L67-L69